raspberry/bzzz/sensors/pressure_sensor.py

```diff
@@ -1,210 +1,221 @@
-    def _get_current_pressure_measurement(self):
-        self.__readBmp180()
-        pressure_reading = self.__pressure
-        return pressure_reading
-    def _update_pressure(self):
-        with self.__lock:
-            # Get current pressure measurement and add to readings list
-            pressure_reading = self._get_current_pressure_measurement()
-            self.__pressure_readings_list.append(pressure_reading)
-            # Remove the oldest reading to maintain the buffer size
-            if len(self.__pressure_readings_list) > self.window_length:
-                self.__pressure_readings_list.pop(0)
-            # Use the data processor to calculate the current pressure
-            self.__current_pressure = self.data_processor.process(self.__pressure_readings_list)
-    def _calculate_altitude_from_pressure(self):
-        with self.__lock:
-        # check calculate altitude from pressure
-            self.__current_altitude = 44330 * \
-                (1 - (self.__current_pressure/self.reference_pressure_at_sea_level)**(1/5.255))
-    
-    def update_altitude(self):
-        self._update_pressure()
-        self._calculate_altitude_from_pressure()
-        if self.log_file is not None:
-    
-            current_timestamp = datetime.datetime.now()
-            data_to_log = (self.pressure, self.altitude)
-            self.logger.record(current_timestamp, data_to_log)
-            time.sleep(0.5)
-   
-
+        B3 = (((AC1 * 4 + X3) * 2) + 2) / 4.0
+        X1 = AC3 * B6 / 8192.0
+        X2 = (B1 * (B6 * B6 / 2048.0)) / 65536.0
+        X3 = ((X1 + X2) + 2) / 4.0
+        B4 = AC4 * (X3 + 32768) / 32768.0
+        B7 = ((UP - B3) * (25000.0))
+        pressure = 0.0
+        if B7 < 2147483648 :
+            pressure = (B7 * 2) / B4
+        else :
+            pressure = (B7 / B4) * 2
+        X1 = (pressure / 256.0) * (pressure / 256.0)
+        X1 = (X1 * 3038.0) / 65536.0
+        X2 = ((-7357) * pressure) / 65536.0
+        pressure = (pressure + (X1 + X2 + 3791) / 16.0)
+        return pressure, Temperature
+              
+            
+    def __initialise(self, num_initial_readings=30):
+        """
+        Initialise the altitude by averaging the first few readings.
+        :param num_initial_readings: Number of readings to average for initial altitude.
+        """        
+        sum_pressures = 0
+        do_run = True
+        idx_valid_measurements = 0
+        idx_attempts = 0
+        max_attempts = 4*num_initial_readings
+        time.sleep(3)
+        while do_run:
+            p = self.pressure()
+            if not np.isnan(p):
+                sum_pressures += p
+                idx_valid_measurements += 1
+            if idx_valid_measurements == num_initial_readings:
+                break
+            time.sleep(0.2)
+            idx_attempts += 1
+            if idx_attempts >= max_attempts:
+                raise Exception("Barometer initialisation took too long (too many nan's)")
```

Comment:
Does this happen in practice?

Reply:
no it doesn't, I believe this was happening before I had fixed the code. it has never thrown that exception when testing since.